### DISCUSSION TITLE
How do I upgrade the Python version used by existing project?

### DISCUSSION THREAD

**Discussion by anentropic** [pdm-project/pdm#3172] [Q&A]
My project was on `==3.11.5` and that is what is in the current lock file

I want to upgrade to 3.11.10

I did:
```
pdm use 3.11.10
pdm lock
pdm install
```

but the install fails because the lock is still stuck on `==3.11.5`:
```
╰─ pdm install
ERROR: None of the lock targets matches the current env (==3.11.10, macos_14_0_arm64, cpython):
 - (==3.11.5, macos_14_0_arm64, cpython)
[PdmException]: No compatible lock target found
WARNING: Add '-v' to see the detailed traceback
```

**Comment by o-moe** [comment]
Try specifying the new python version when locking with `pdm lock --python=3.11.10` (or probably delete the lock file and `pdm install` again)
